spec/verify_spec.rb

```diff
@@ -1,42 +1,56 @@
 # frozen_string_literal: true
 
 describe 'Verify' do
```

Comment:
I understand if you want to keep the test structure consistent with the other tests, however, rspec is such a powerful testing framework to write expressive tests, so I'd recommend:

* use `describe` to describe features
* use `context` to describe the feature in different situations/contexts
* use `it` to specify expectations (one `it` per `expect`)

For example:

```ruby

describe 'Verify an email OTP' do
  context 'using an existing OTP' do
    ...

    it 'has the ID verify-id' do
      expect(verify.id).to eq 'verify-id'
    end

    it 'has the verified status' do
      expect(verify.status).to eq 'verified'
    end
  end

  context 'using a non-existing OTP' do
    it 'raises an error' do
      expect { @client.verify('verify-id') }.to raise_error(MessageBird::ErrorException)
    end
  end

  context "<insert here any other scenario>" do
    ...
  end
end
```

Reply:
Agreed, too late 😆 .

For the topmost `describe` block I usually prefer to use the class I'm testing, it gives me an instance of the class in the `subject` variable already to be used in the examples, and I can access it in some magic ways:

```ruby
describe Array do
  context "new instances" do
    it { should be_empty }
    # or
    it { is_expected.to be_empty }
  end
end
```